src/attr/incomparable.rs

```diff
@@ -0,0 +1,52 @@
+//! Attribute parsing for the `default` option.
+
+use proc_macro2::Span;
+use syn::{spanned::Spanned, Meta, Result};
+
+use crate::{attr::DeriveTrait, DeriveWhere, Error};
+
+/// Stores if this variant should be the default when implementing
+/// [`Default`](std::default::Default).
+#[derive(Clone, Copy, Default)]
+#[cfg_attr(test, derive(Debug))]
+pub struct Incomparable(pub Option<Span>);
+
+impl Incomparable {
+	/// Token used for the `default` option.
+	pub const INCOMPARABLE: &'static str = "incomparable";
+
+	/// Adds a [`Meta`] to this [`Default`](self).
+	pub fn add_attribute(&mut self, meta: &Meta, derive_wheres: &[DeriveWhere]) -> Result<()> {
+		debug_assert!(meta.path().is_ident(Self::INCOMPARABLE));
+
+		if let Meta::Path(path) = meta {
+			if self.0.is_some() {
+				Err(Error::option_duplicate(path.span(), Self::INCOMPARABLE))
+			} else {
+				let mut impl_cmp = false;
+
+				for trait_ in derive_wheres
+					.iter()
+					.flat_map(|derive_where| derive_where.traits.iter())
+				{
+					match trait_ {
+						DeriveTrait::Eq | DeriveTrait::Ord => {
+							return Err(Error::non_partial_incomparable(path.span()));
+						}
+						DeriveTrait::PartialEq | DeriveTrait::PartialOrd => impl_cmp = true,
+						_ => {}
```

Comment:
`incomparable` is only implemented for `PartialEq`, so I think `Ord` and `PartialOrd` should not be considered here, right?

Reply:
Well they have to, as PartialEq has to match PartialOrd. Just didn't implement that yet (I had a task in the pr description for that)